[BUG] Windows Build Failing: Could not resolve open-nextserver-functionsdefaultcache.cjs

### Describe the bug

When attempting to deploy the [firelabs-net/workers-next](https://github.com/flarelabs-net/workers-next) template and having made no changes to the template, the `npm run deploy` fails.  Specifically the `wrangler deploy` step fails. 

This is happening on a Windows machine in standard PowerShell terminal as well as WSL Ubuntu and Debian. 

```
✘ [ERROR] Build failed with 1 error:

  ✘ [ERROR] Could not resolve
  "C:Users\tdesmondProjects\tempempty-hall-5379.open-nextserver-functionsdefaultcache.cjs"

      .open-next/server-functions/default/handler.mjs:30791:39:
        30791 │ ...= require('C:\Users\tdesmond\Projects\temp\empty-hall-5379\....
              ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    You can mark the path
  "C:Users\tdesmondProjects\tempempty-hall-5379.open-nextserver-functionsdefaultcache.cjs" as
  external to exclude it from the bundle, which will remove this error. You can also surround this
  "require" call with a try/catch block to handle this failure at run-time instead of bundle-time.
``` 

### Steps to reproduce

1. On a Windows machine create new Next.js project based on the 0.3 build `npm create cloudflare@latest -- --template https://github.com/flarelabs-net/workers-next`
2. Attempt to deploy the application `npm run deploy`

### Expected behavior

I'm expecting `npm run deploy` to complete without errors and deploy my application to Workers Assets. 

### @opennextjs/cloudflare version

0.3

### Wrangler version

3.99.0

### next info output

```shell
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 32206
  Available CPU cores: 22
Binaries:
  Node: 20.16.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.5 // An outdated version detected (latest is 15.1.2), upgrade is highly recommended!
  eslint-config-next: 14.2.5
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.7.2
Next.js Config:
  output: N/A
 ⚠ An outdated version detected (latest is 15.1.2), upgrade is highly recommended!
   Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
   Read more - https://nextjs.org/docs/messages/opening-an-issue
```


### Additional context

_No response_

--- a/.changeset/slow-plants-boil.md
+++ b/.changeset/slow-plants-boil.md
@@ -1,0 +1,5 @@
+---
+"@opennextjs/cloudflare": patch
+---
+
+fix: invalid paths in windows bundles.

--- a/packages/cloudflare/package.json
+++ b/packages/cloudflare/package.json
@@ -72,7 +72,7 @@
   },
   "dependencies": {
     "@dotenvx/dotenvx": "catalog:",
-    "@opennextjs/aws": "https://pkg.pr.new/@opennextjs/aws@688",
+    "@opennextjs/aws": "https://pkg.pr.new/@opennextjs/aws@695",
     "glob": "catalog:",
     "ts-morph": "catalog:",
     "enquirer": "^2.4.1"

--- a/packages/cloudflare/src/cli/build/bundle-server.ts
+++ b/packages/cloudflare/src/cli/build/bundle-server.ts
@@ -8,6 +8,7 @@ import { build, Plugin } from "esbuild";
 
 import { Config } from "../config.js";
 import * as patches from "./patches/index.js";
+import { normalizePath } from "./utils/index.js";
 
 /** The dist directory of the Cloudflare adapter package */
 const packageDistDir = path.join(path.dirname(fileURLToPath(import.meta.url)), "../..");
@@ -117,10 +118,9 @@ globalThis.__BUILD_TIMESTAMP_MS__ = ${Date.now()};
 
   const isMonorepo = monorepoRoot !== appPath;
   if (isMonorepo) {
-    const packagePosixPath = packagePath.split(path.sep).join(path.posix.sep);
     fs.writeFileSync(
       path.join(outputPath, "handler.mjs"),
-      `export * from "./${packagePosixPath}/handler.mjs";`
+      `export * from "./${normalizePath(packagePath)}/handler.mjs";`
     );
   }
 
@@ -183,7 +183,7 @@ function createFixRequiresESBuildPlugin(config: Config): Plugin {
     name: "replaceRelative",
     setup(build) {
       // Note: we (empty) shim require-hook modules as they generate problematic code that uses requires
-      build.onResolve({ filter: /^\.\/require-hook$/ }, () => ({
+      build.onResolve({ filter: /^\.(\/|\\)require-hook$/ }, () => ({
         path: path.join(config.paths.internal.templates, "shims", "empty.js"),
       }));
     },

--- a/packages/cloudflare/src/cli/build/open-next/createServerBundle.ts
+++ b/packages/cloudflare/src/cli/build/open-next/createServerBundle.ts
@@ -17,6 +17,8 @@ import { openNextReplacementPlugin } from "@opennextjs/aws/plugins/replacement.j
 import { openNextResolvePlugin } from "@opennextjs/aws/plugins/resolve.js";
 import type { FunctionOptions, SplittedFunctionOptions } from "@opennextjs/aws/types/open-next.js";
 
+import { normalizePath } from "../utils/index.js";
+
 export async function createServerBundle(options: buildHelper.BuildOptions) {
   const { config } = options;
   const foundRoutes = new Set<string>();
@@ -206,7 +208,7 @@ async function generateBundle(
       outfile: path.join(outputPath, packagePath, `index.${outfileExt}`),
       banner: {
         js: [
-          `globalThis.monorepoPackagePath = "${packagePath}";`,
+          `globalThis.monorepoPackagePath = "${normalizePath(packagePath)}";`,
           name === "default" ? "" : `globalThis.fnName = "${name}";`,
         ].join(""),
       },
@@ -273,10 +275,10 @@ function addMonorepoEntrypoint(outputPath: string, packagePath: string) {
   //       the root of the bundle. We will create a dummy `index.mjs`
   //       that re-exports the real handler.
 
-  // TOOD: use helper
-  // Always use posix path for import path
-  const packagePosixPath = packagePath.split(path.sep).join(path.posix.sep);
-  fs.writeFileSync(path.join(outputPath, "index.mjs"), `export * from "./${packagePosixPath}/index.mjs";`);
+  fs.writeFileSync(
+    path.join(outputPath, "index.mjs"),
+    `export * from "./${normalizePath(packagePath)}/index.mjs";`
+  );
 }
 
 async function minifyServerBundle(outputDir: string) {

--- a/packages/cloudflare/src/cli/build/patches/investigated/patch-cache.ts
+++ b/packages/cloudflare/src/cli/build/patches/investigated/patch-cache.ts
@@ -2,6 +2,8 @@ import path from "node:path";
 
 import type { BuildOptions } from "@opennextjs/aws/build/helper.js";
 
+import { normalizePath } from "../../utils/index.js";
+
 /**
  * Sets up the OpenNext cache handler in a Next.js build.
  *
@@ -24,7 +26,7 @@ export async function patchCache(code: string, openNextOptions: BuildOptions): P
   return code.replace(
     "const { cacheHandler } = this.nextConfig;",
     `const cacheHandler = null;
-CacheHandler = require('${cacheFile}').default;
+CacheHandler = require('${normalizePath(cacheFile)}').default;
 `
   );
 }

